### PR TITLE
[#3180] - add relations and funding_agencies updating via rest api

### PR DESCRIPTION
**Pull request by sblack-usu**
### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [x] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. Add funding_agencies and relations via the rest api (try using swagger!)

### CODE REVIEW

**Comment by zhiyuli** [comment]
@sblack-usu 
I tried this branch with hs_restclient locally and I was able to create funding_agencies and relations. 
The extra_metadata will be fixed in a separate branch?

**Comment by cuahsimarko** [comment]
@zhiyuli can you complete this review, as I've spent all my review time on frontend stories as well as the performance one.

**Comment by sblack-usu** [comment]
I'll take a look at how extra_metadata works and see if it's easy enough to wrap into this PR.  I'm also adding the asserts you suggested in the tests.

File: hs_core/tests/api/rest/test_resource_science_meta.py [hunk 1]
```diff
@@ -100,11 +100,35 @@ def test_put_scimeta_generic_resource(self):
...
+                     "award_number": "NSF-123",
+                     "agency_url": "https://www.google.com",
+                 }
+            ]
         }
         response = self.client.put(sysmeta_url, put_data, format='json')
         self.assertEqual(response.status_code, status.HTTP_202_ACCEPTED)
         self.assertEqual(self.resource.metadata.dates.all().count(), 3)
         self.assertEqual(self.resource.metadata.sources.all().count(), 2)
+        self.assertEqual(self.resource.metadata.relations.all().count(), 2)
```
[the remark this answers is not in the document]
Comment (sblack-usu) [reply hs_core/tests/api/rest/test_resource_science_meta.py]: done

**Comment by sblack-usu** [comment]
@zhiyuli - extra metadata endpoint already exists and is setup on hs_restclient (scimeta.custom)

**Comment by zhiyuli** [comment]
@sblack-usu 
Tested the GET endpoint for extra metadata and it worked for me.
A new function would be needed in hs_restclient.

**Pull request merged**